### DISCUSSION TITLE
Checking for Resistance

### DISCUSSION THREAD

**Discussion by g-qwz** [pybricks/support#1486] [Q&A]
Is there a way with pybricks v2.0 (ev3) to check if the Motor gets resistance? (e.g. by checking if it can't move anymore)

**Comment by laurensvalk** [comment]
You can check `motor.control.stalled()`, which returns `True` when stalled.

In newer releases (Powered Up) this was later moved to `motor.stalled()`.

**Reply by jaguilar** [reply]
One of the things I noticed about run_until_stalled when I was using it before is that it pushes pretty hard even after it reaches the end of the model's allowable throw. Is there any way to make stall detection more sensitive? I was thinking about something like this for my next project. Maybe the motor could be waggled a little in both directions to see in which direction the motor moves most freely.

Once you've determined which direction allows the freest movement, then you'd run the motor in that direction using the minimum possible duty cycle that gives a reasonably steady pace of movement. Then you would try to anomaly detect to see when the speed slows down. Then, rather than running for 500ms past the point of slowdown, perhaps the motor could be stopped within a few tens of milliseconds. 

(Although, perhaps this whole thing is not possible because of a lack of resolution on the motor angle.)

**Reply by laurensvalk** [reply]
> One of the things I noticed about run_until_stalled when I was using it before is that it pushes pretty hard even after it reaches the end of the model's allowable throw. 

That's what the `duty_limit` is for:

![image](https://github.com/pybricks/support/assets/12326241/9a0cbd2e-3025-450c-ad60-810b67f30413)

**Reply by jaguilar** [reply]
Yes, I was using the minimum duty_limit that would induce motor movement.

(Now, it may be that this stuff worked less well in pybricks 2.0. I haven't tried on a recent release, since I only have EV3.)
